content/08-mongodb/14-mongodb-encryption.mdx

```diff
@@ -0,0 +1,94 @@
+- [Asymmetric Encryption](https://en.wikipedia.org/wiki/Public-key_cryptography) - A method also known as public key cryptography that creates two different keys, a private key and a public key that are not identical. The public key can be shared with anyone, and the private key only with authorized people. The public key can be used to encrypt data that can only be decrypted by the private key.
+
+Both methods have some inherent vulnerabilities depending on the data status but cover up RBAC missed issues like networking snooping and data theft. This assurance is why MongoDB employs both methods across its databases, depending on the data state.  
+
+### Encryption in transit
+[Encryption in transit](/intro/database-glossary#encryption-transport) is securing data when it is in motion from one point to another. Before being sent, the data is encrypted for its “journey” and then decrypted and verified at the reception point. The best visualization is to imagine the data as money transported via an armored van from one bank branch to another to replenish the vault. 
+
+[MongoDB Atlas](https://www.prisma.io/dataguide/mongodb/mongodb-atlas-setup) supports [transport layer security (TSL)](https://en.wikipedia.org/wiki/Transport_Layer_Security) and secure socket layer (SSL) for communications between application client and server and within intra-cluster communications by setting certificates for the servers. A certificate is an electronic document used to prove the validity of ownership of a private key. Certificates ensure data is encrypted for transport over a trusted network connection preventing network snooping activities such as packet sniffing or IP/DNS spoofing. 
+
+With the inclusion of a public key, we see that MongoDB utilizes asymmetric encryption to secure data in a state of motion. Symmetric encryption would require the encryption key to be communicated between parties over some channel. The single key must remain secure during the transmission and is open to risk at this stage. Asymmetric encryption theoretically solves this insecure transmission problem with its public keys. Encrypted communication can take place without any prior communication.
+
+By encrypting with a public key, the sender’s message can only be decrypted by the intended recipient with their private key. Public key cryptography is advantageous in transit because there is no need to ever share private keys and the use of signatures ensures a recipient can verify that a message comes from a trusted sender. 
+
+The downside to asymmetric encryption is that is does take longer than symmetric encryption and requires more resource consumption to achieve. If your data is not actively moving from device to device or network-to-network, then these tradeoffs are not necessary to undertake and symmetric encryption provides ample security.
```

Comment:
I'm not sure how much we need to go into this, but one comment I have is that the asymmetric keys are only used to establish the connection securely, so the asymmetric overhead is a fixed cost.  Once that is done (both parties trust each other, identity has been confirmed, whatever is outlined in the connection parameters you're using) since the asymmetric keys are used to create a symmetric key that is used for the actual data for the remainder of the connection, the overhead drops off.

So basically: the overhead of asymmetric encryption only applies once per connection (it isn't a cost you pay for every message over that connection).  As I said, I'm not sure if we need to go into that, but I wanted to clarify in case that changes how we talk about the relative cost / benefits of the asymmetric keys a bit.